spark/src/main/scala/com/alibaba/graphar/GraphInfo.scala

```diff
@@ -203,6 +203,34 @@ class GraphInfo() {
   @BeanProperty var vertices = new java.util.ArrayList[String]()
   @BeanProperty var edges = new java.util.ArrayList[String]()
   @BeanProperty var version: String = ""
+
+  var vertexInfos: Map[String, VertexInfo] = Map[String, VertexInfo]()
+  var edgeInfos: Map[String, EdgeInfo] = Map[String, EdgeInfo]()
+
+  def addVertexInfo(vertexInfo: VertexInfo): Unit = {
+    vertexInfos += (vertexInfo.getLabel -> vertexInfo)
+  }
+
+  def addEdgeInfo(edgeInfo: EdgeInfo): Unit = {
+    edgeInfos += (edgeInfo.getConcatKey() -> edgeInfo)
+  }
+
+  def getVertexInfo(label: String): VertexInfo = {
+    vertexInfos(label)
+  }
+
+  def getEdgeInfo(srcLabel: String, edgeLabel: String, dstLabel: String): EdgeInfo = {
+    val key = srcLabel + GeneralParams.regularSeperator + edgeLabel + GeneralParams.regularSeperator + dstLabel
+    edgeInfos(key)
+  }
+
+  def getVertexInfos(): Map[String, VertexInfo] = {
```

Comment:
we could modify the implementation of GraphTransformer to use the maps in GraphInfo, avoiding to construct the info maps again

Reply:
Fixed.